workspace/__ardulib__/Graphics/Image.h

```diff
@@ -0,0 +1,42 @@
+
+#ifndef IMAGE_H
+#define IMAGE_H
+
+#ifndef PGM_SPACE
+#ifdef __AVR__
+#include <avr/pgmspace.h>
+#elif defined(ESP8266) || defined(ESP32)
+#include <pgmspace.h>
+#endif // __AVR__
+#endif // PGM_SPACE
+
+/*
+ * Not all non-AVR boards installs define macros
+ * for compatibility with existing PROGMEM-reading AVR code.
+ */
+#ifndef pgm_read_byte
+#define pgm_read_byte(addr) (*(const unsigned char *)(addr))
+#endif
+
+#include "XGraphics.h"
+
+class Image : public XGraphics {
+private:
+    BBox _imageBBox;
+
+    uint8_t* _bitmapBuffer;
```

Comment:
It looks like pointer to the buffer, width, and height stored twice.
The first time in the `Bmp` struct, the second time in the instances of Image.

I'm not sure, but probably we can replace them with the single pointer to the `Bmp` struct? And rename `setBitmap` method to `linkBitmap` :)

Reply:
It is done for tiling or displaying only a part of image.
For example the size of the source bitmap is fixed 50x50. Image field is 100x100. Image should tile to four bitmaps for both axes.

Reply:
Em... I told about `_bitmapWidth` and `_bitmapHeight`, which is always equal to `bitmap->width` and `bitmap-height`. So I think we don't need to copy their values here. The image field is stored in the `_imageBBox`, so it used for tiles / cropping the bitmap. Right?

So we don't need to store `_bitmapWidth` twice, I think, because `_bitmapWidth` always equals to `bitmap->width`.